### ISSUE TITLE
the implementation is not as clean as it could be due to upstream issues in terraform

### ISSUE THREAD

**Issue by samstav** [samstav/tf_mailgun_aws#2]
https://github.com/hashicorp/terraform/issues/5367#issuecomment-264409516

**Comment by phillbaker** [comment]
For what it's worth, I've been trying to put together a terraform mailgun provider as a plugin that uses v3 of the api and deals with some of these short comings: https://github.com/phillbaker/terraform-provider-mailgunv3.

**Comment by samstav** [comment]
@phillbaker nice! I would love to use your provider instead. Are you going to submit a PR to hashicorp/terraform ?

**Comment by phillbaker** [comment]
@samstav yup, it already supports domains and routes: https://github.com/phillbaker/terraform-provider-mailgunv3#usage.

I've been meaning to finish up the tests, so probably need to do that before getting it into a PR, but have been using it regularly.
